test/cartesian/getEquidistantTicks.spec.ts

```diff
@@ -0,0 +1,40 @@
+// unit test for getEquidisantTicks
+
+import _ from 'lodash';
+import { getEquidistantTicks } from '../../src/cartesian/getEquidistantTicks';
+import { CartesianTickItem } from '../../src/util/types';
+
+describe('getEquidistantTicks', () => {
+  // We mock getting the tick width by simply using the value of the tick.
+  // This makes testing rather easy and intuitive.
+  const getTickSize = (tick: CartesianTickItem) => {
+    return tick.value;
+  };
+
+  it('should return empty array if no ticks are passed', () => {
+    const result = getEquidistantTicks(1, { start: 0, end: 100 }, getTickSize, [], 0);
+    expect(result).toEqual([]);
+  });
+
+  test.each([
+    // If all ticks fit, we show all ticks
+    { ticksThatFit: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], resultingTicks: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] },
+    { ticksThatFit: [0, 1, 2, 3, 4, 5, 6, 7, 8], resultingTicks: [0, 2, 4, 6, 8] },
+    { ticksThatFit: [0, 1, 2, 3, 4, 5, 6, 7], resultingTicks: [0, 5] },
+    { ticksThatFit: [0, 1, 2, 3, 4, 5, 6], resultingTicks: [0, 5] },
+    { ticksThatFit: [0, 1, 2, 3, 4, 5], resultingTicks: [0, 5] },
+    { ticksThatFit: [0, 1, 2, 3, 4], resultingTicks: [0] },
```

Comment:
This section deserves scrutiny. Is this the behaviour we want to achieve?

Reply:
is this really true in the browser though?

I feel like all of these should be able to fit? every time?

I kind of think we shouldn't always preserve start and instead show as many as we can while maintaining equidistance

Reply:
I construct the ticks in the test such that this is true in the browser. 
But good point, if we were to dynamically allow starting from any tick but still only show equidistant ticks, we could show much more.

Reply:
Really, we could consider to redo the interface completely. As far as I understand it, the options currently are: 

alwaysShowStart | alwaysShowEnd | collisionHandlingStrategy | currentName
--- | --- | --- | ---- 
true | false | keep left | preserveStart
false | true | keep right | preserveEnd 
true | true | keep left | preserveStartEnd
true | false | compute collision to direct neighbour and then select every nTH | equidistantPreserveStart 
true | false | show every nTH without collision | equidistant

We could have the option to choose `equidistant` without `alwaysShowStart`.

I admittedly find the naming of these strategies not that insightful. We should probably investigate how others do this.

Reply:
The more I think about this, I believe we should simply replace the existing `equidistantPreserveStart` with this implementation. 

This would be a breaking change, as it changes existing behaviour, but the change would only be to the better, namely we would show more ticks than we previously did, and there would yet be no collision.

@ckifer What do you think about this approach? 

I like it, because it does not further complicate the API and it improves the existing behaviour (almost a bug fix)

Reply:
I'm on board - of people complain that their chart looks better then... not sure lol. If anything we could do the same thing with preserveEnd if we wanted no?

Reply:
Done. Thanks for the discussion.

Reply:
> I'm on board - of people complain that their chart looks better then... not sure lol. If anything we could do the same thing with preserveEnd if we wanted no?

They will look better, Coltin! :P

Reply:
Exactly 🚀. If there are complaints I will say but... they look better 😂